### DISCUSSION TITLE
How to insert (with javascript) html + property, inside mavo app?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#849] [Q&A]
Hi

I'm trying to create a simple javascript to insert a html element that 'declares' a new property that mavo can edit

I've done this [codepen](https://codepen.io/galinhalx/pen/eYVBwmZ?editors=1010)

(Sorry I'm not a javascript expert)

Is there any demo that have this usecase?

As I have too much properties in my page, I'm trying to cut my main page into small parts, and load them when needed, so
I'm trying to have a button, outside the mavo app, that creates element & property, that mavo recognises.

Maybe it should use hooks like #536 , but can't figure how to use it

**Comment by LeaVerou** [comment]
Unfortunately this is not possible yet. The only way to use Mavo to make Mavos with entirely dynamic properties is to use an `<iframe>` with expressions in its `srcdoc`. If it's only about adding and removing predefined properties you could also just use `mv-if`.

**Comment by GalinhaLX** [comment]
Hi @LeaVerou ,


I understood `srcdoc` is an alternative to `src`, in case `<iframe>` is not supported by browser, 
so I'm sorry, but I don't understand what you mean with "`<iframe>` with expressions in its `srcdoc`." 

I tried something like

```
<iframe srcdoc="<p property='foo' >Hello world!</p>"></iframe>


<p id="output">My message is [foo]</p>
```

Can't make it work :(

- text "Hello world" is displayed in `<iframe>` , but not editable
- text is not displayed in innerHtml of` id="output"`

---------

Just realise you're talking about expressions and not properties.

But now I'm confused: how to set a value to a property with an expression? (I see how to set it with mv-action=set(property, value) only)
